tests/components/freebox/test_binary_sensor.py

```diff
@@ -0,0 +1,32 @@
+"""Tests for the Freebox sensors."""
+from unittest.mock import Mock
+
+from homeassistant.components.freebox.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .const import DATA_STORAGE_GET_RAIDS, MOCK_HOST, MOCK_PORT
+
+from tests.common import MockConfigEntry
+
+
+def bool_to_binary_sensor(value):
+    """Convert the mock data (bool) to the expected BinarySensor state string."""
+    return "on" if value else "off"
+
+
+async def test_raid_array_degraded(hass: HomeAssistant, router: Mock) -> None:
+    """Test raid array degraded binary sensor."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_HOST: MOCK_HOST, CONF_PORT: MOCK_PORT},
+        unique_id=MOCK_HOST,
+    )
+    entry.add_to_hass(hass)
+    assert await async_setup_component(hass, DOMAIN, {})
+    await hass.async_block_till_done()
+
+    assert hass.states.get(
+        "binary_sensor.freebox_server_r2_raid_array_0_degraded"
+    ).state == bool_to_binary_sensor(DATA_STORAGE_GET_RAIDS[0]["degraded"])
```

Comment:
```suggestion
    assert hass.states.get(
        "binary_sensor.freebox_server_r2_raid_array_0_degraded"
    ).state == "off"

    # Now simulate we degraded
    DATA_STORAGE_GET_RAIDS_DEGRADED = deepcopy(DATA_STORAGE_GET_RAIDS)
    DATA_STORAGE_GET_RAIDS_DEGRADED[0]["degraded"] = True
    router().storage.get_raids.return_value = DATA_STORAGE_GET_RAIDS_DEGRADED

    # Simulate an update
    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=60))
    # To execute the save
    await hass.async_block_till_done()

    assert hass.states.get(
        "binary_sensor.freebox_server_r2_raid_array_0_degraded"
    ).state is "on"
```

Reply:
Awesome, thanks ! I really didn't know how to do this, now i do ;)